src/libraries/Microsoft.PowerFx.Connectors/Environment/TabularService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public abstract class TabularService
+    {
+        public TableType TableType { get; private set; } = null;
+
+        public bool IsInitialized => TableType != null;
+
+        public virtual ConnectorTableValue GetTableValue()
+        {
+            return IsInitialized
+                ? new ConnectorTableValue(this, TableType)
+                : throw new InvalidOperationException("Tabular service is not initialized.");
+        }
+
+        public async Task InitAsync(CancellationToken cancellationToken)
```

Comment:
> public async Task InitAsync(CancellationToken cancellationToken) [](http://example.com/codeflow?start=7&length=65)

tbd?

What we really want is an async constructor

Reply:
It can be sync and async 
See File tabular example which is sync